### PR TITLE
Add alternative code paths depending on parameters

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Add alternative code paths depending on parameters which allows for coming changes to common tools.
This change is solely intended to prevent breaking the build while making changes to common across various repos.

Sample Release Build: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=756200&view=results

### CODE REVIEW

File: eng/scripts/Language-Settings.ps1 [hunk 1]
```diff
@@ -6,28 +6,65 @@ $packagePattern = "*.nupkg"
...
-    return $pkgProp
   }
-  else
+  else 
   {
-    return $null
+    $pkgProperties = @()
+    if (Test-Path (Join-Path $pkgDirectoryPath "src"))
+    {
+      $projectFilePaths = Get-ChildItem (Join-Path $pkgDirectoryPath "src") "*.csproj" -File
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:47]: I think it is save to assert that there is only one csproj here and if we find more then one we should error/warn and only take the first.

File: eng/scripts/Language-Settings.ps1 [hunk 2]
```diff
@@ -6,28 +6,65 @@ $packagePattern = "*.nupkg"
 $MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/dotnet-packages.csv"
 $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=dotnet%2F&delimiter=%2F"
 
-function Get-dotnet-PackageInfoFromRepo ($pkgPath, $serviceDirectory, $pkgName)
+function Get-dotnet-PackageInfoFromRepo
 {
-  $projectPath = Join-Path $pkgPath "src" "$pkgName.csproj"
-  if (Test-Path $projectPath)
+  [CmdletBinding()]
+  Param(
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:12]: Lets not change these parameters yet until we make the changes in common.

File: eng/scripts/Language-Settings.ps1 (hunk 1 above)
Comment (weshaggard) [reply eng/scripts/Language-Settings.ps1:47]: Once we do that we should be able to reuse most of this function.

Comment (chidozieononiwu) [reply eng/scripts/Language-Settings.ps1:47]: Again am using different argument names inthe different code blocks. Also what if the first .csproj is the wrong one?

File: eng/scripts/Language-Settings.ps1 [hunk 3]
```diff
@@ -8,9 +8,31 @@ $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=con
...
+  if ($pkgName)
   {
+    $projectPath = Join-Path $projDirPath "$pkgName.csproj"
+  }
+  else
+  {
+    $projectPath = (Resolve-Path (Join-Path $projDirPath "*.csproj")).path
+  }
+
+  if ($projectPath.Count -gt 1)
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:27]: This will fail once we turn on strict mode because you are essentially changing types. I suggest using a different variable name and doing this inside the else block above.

File: sdk/mediaservices/ci.yml [hunk 4]
```diff
@@ -15,7 +15,7 @@ pr:
   branches:
     include:
       - master
-    - main
+      - main
```
Comment (mitchdenny) [inline sdk/mediaservices/ci.yml:18]: This change seems unrelated to the PR?

File: eng/scripts/Language-Settings.ps1 (hunk 1 above)
Comment (mitchdenny) [reply eng/scripts/Language-Settings.ps1]: I think the assertion here is that if we have two ```*.csproj``` files in the one directory here that it is a failure condition. That shouldn't happen.

**Review by mitchdenny — CHANGES_REQUESTED** [review]
Would be good to remove that seemingly unrelated ```ci.yml``` change. Was this an indentation from @sima-zhu's change?

File: eng/scripts/Language-Settings.ps1 [hunk 5]
```diff
@@ -8,9 +8,35 @@ $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=con
...
+    else {
+      $projectPath = $projectPaths
+    }
+  }
+
+
+
+  if ($projectPath -and (Test-Path $projectPath))
   {
+    $pkgName = Split-Path -Path $projectPath -LeafBase
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:39]: While this is the default package name it is possible for projects to override this in the project itself. I don't think we have any such cases currently but it is something to think about in the future. One of the reasons it might happen for example is if we hit long path issues.

File: sdk/mediaservices/ci.yml (hunk 4 above)
Comment (weshaggard) [reply sdk/mediaservices/ci.yml:18]: I agree they seem unrelated but perhaps you hit an issue with this pipeline while testing. In any case we need to fix this and make sure this problem isn't repeated in other ci.yml files cc @sima-zhu

**Review by weshaggard — APPROVED** [review]

File: eng/scripts/Language-Settings.ps1 (hunk 5 above)
Comment (chidozieononiwu) [reply eng/scripts/Language-Settings.ps1:39]: Actually you are right. This will be bad for cognitiveservices where we did exactly that.

File: sdk/mediaservices/ci.yml (hunk 4 above)
Comment (chidozieononiwu) [reply sdk/mediaservices/ci.yml:18]: The change is actually a fix, and is intended else it will not be parsed by the yml parser we use.

**Review by mitchdenny — APPROVED** [review]
